docs/cookbook/custom-model.rst

```diff
@@ -0,0 +1,247 @@
+
+This will generate a new migration file which adds the Supplier entity to your database.
+Then update the database using the generated migration:
+
+.. code-block:: bash
+
+    $ php app/console doctrine:migrations:migrate
+
+4. Add ResourceInterface to your model class
+--------------------------------------------
+
+Go to the generated class file and make it implement the ``ResourceInterface``:
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Entity;
+
+    use Sylius\Component\Resource\Model\ResourceInterface;
+
+    class Supplier implements ResourceInterface
+    {
+        // ...
+    }
+
+5. Register your entity as a Sylius resource
+--------------------------------------------
+
+If you don't have it yet create a file ``app/config/resources.yml``, import it in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    imports:
+        - { resource: "resources.yml" }
+
+And add these few lines in the ``resources.yml`` file:
+
+.. code-block:: yaml
+
+    # app/config/resources.yml
+    sylius_resource:
+        resources:
+            app.supplier:
+                driver: doctrine/orm
+                classes:
+                    model: AppBundle\Entity\Supplier
+
+To check if the process was run correctly run such a command:
+
+.. code-block:: bash
+
+    $ php app/console debug:container | grep supplier
+
+The output should be:
+
+.. image:: ../_images/container_debug_supplier.png
+    :align: center
+
+6. Use Sylius API to create new resource
+----------------------------------------
+
+Creating OAuth client:
+
+.. code-block:: bash
+
+    $ php app/console sylius:oauth-server:create-client --grant-type="password" --grant-type="refresh_token" --grant-type="token"
+
+It will give you such a response:
+
+.. code-block:: bash
+
+    A new client with public id XYZ, secret ABC has been added
+
+Run your application on a built-in server:
+
+.. code-block:: bash
+
+    $ php ap/console server:start localhost:8000
+
+To obtain authorization token for the ``api@example.com`` user with password ``sylius-api`` run:
+
+.. code-block:: bash
+
+    $ curl http://localhost:8000/api/oauth/v2/token -d "client_id"=XYZ -d "client_secret"=ABC -d "grant_type"=password -d "username"=api@example.com -d "password"=sylius-api
+
+This will give you such a response:
+
+.. code-block:: bash
+
+    {"access_token":"DEF","expires_in":3600,"token_type":"bearer","scope":null,"refresh_token":"GHI"}
+
+Use the ``access_token`` in the request that will create a new Supplier.
+
+.. code-block:: bash
+
+    $ curl -i -X POST -H "Content-Type: application/json" -H "Authorization: Bearer DEF" -d '{"name": "Example", "description": "Lorem ipsum", "enabled": true}' http://localhost:8000/api/suppliers/
+
+.. tip::
+
+    Read more about authorizing in API :doc:`here </api/authorization>`. Different requests to Sylius API are described in `the Lionframe docs <http://lakion.com/lionframe>`_.
```

Comment:
I think we should add examples of getting a list of suppliers and partial update of a supplier.